src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -44,30 +47,93 @@ public IEnumerable<DataSource> ListAllDataSources()
 
     /// <summary>
     /// Constructor for runtimeConfig.
+    /// To be used when setting up from cli json scenario.
     /// </summary>
-    /// <param name="Schema">schema.</param>
+    /// <param name="Schema">schema for config.</param>
     /// <param name="DataSource">Default datasource.</param>
     /// <param name="Runtime">Runtime settings.</param>
     /// <param name="Entities">Entities</param>
+    /// <param name="DataSourceFiles">List of datasource files for multiple db scenario.Null for single db scenario.</param></param>
     [JsonConstructor]
-    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities)
+    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities, DataSourceFiles? DataSourceFiles = null)
     {
         this.Schema = Schema;
         this.DataSource = DataSource;
         this.Runtime = Runtime;
         this.Entities = Entities;
-        this._dataSourceNameToDataSource = new Dictionary<string, DataSource>();
-        this._defaultDataSourceName = Guid.NewGuid().ToString();
-        this._dataSourceNameToDataSource.Add(this._defaultDataSourceName, this.DataSource);
+        _defaultDataSourceName = Guid.NewGuid().ToString();
+
+        // we will set them up with default values
+        _dataSourceNameToDataSource = new Dictionary<string, DataSource>
+        {
+            { _defaultDataSourceName, this.DataSource }
+        };
+
+        _entityNameToDataSourceName = new Dictionary<string, string>();
 
-        this._entityNameToDataSourceName = new Dictionary<string, string>();
         foreach (KeyValuePair<string, Entity> entity in Entities)
         {
-            _entityNameToDataSourceName.TryAdd(entity.Key, this._defaultDataSourceName);
+            _entityNameToDataSourceName.TryAdd(entity.Key, _defaultDataSourceName);
+        }
+
+        // Process data source and entities information for each database in multiple database scenario.
+        this.DataSourceFiles = DataSourceFiles;
+
```

Comment:
extra line

Reply:
I feel in this case it makes it more easy to read the code.